### PR TITLE
add logdet and logabsdet

### PR DESCRIPTION
**Pull request by mileslucas**

### CODE REVIEW

File: src/WoodburyMatrices.jl [hunk 1]
```diff
@@ -88,7 +88,16 @@ end
...
+logdet(W::AbstractWoodbury) = logdet(W.A) + logdet(W.C) - logdet(W.Cp)
+function logabsdet(W::AbstractWoodbury)
+    lad_A = logabsdet(W.A)
+    lad_C = logabsdet(W.C)
+    lad_Cp = logabsdet(W.Cp)
+    lad = lad_A[1] + lad_C[1] - lad_Cp[1]
+    s = lad_A[2] * lad_C[2] / lad_Cp[2]
+    return lad, s
+end
+end
```
Comment (timholy) [inline src/WoodburyMatrices.jl]: The extra `end` here terminates the module

**Review by timholy — COMMENTED** [review]
Good idea. Has to have tests, though.

**Comment by mileslucas** [comment]
Yep, I threw this up quickly so I’d remember in the morning- I’ll work on some tests today

File: src/WoodburyMatrices.jl (hunk 1 above)
Comment (mileslucas) [reply src/WoodburyMatrices.jl]: whoopsies- copy and paste introduces a bug that takes me 30 minutes to figure out (part 139587138951)

**Comment by mileslucas** [comment]
okay, I've gotten around to adding some tests. You'll see I've just added some regression tests to make sure we're consistent with `LinearAlgebra`, although it is just copy-paste of the implementation 
> "Our internal investigation has determined no foul play" - Me, a few moments before introducing a bug, probably

If you have any ideas about accuracy tests, I'm open ears; my linalg knowledge is spotty.

File: test/woodbury.jl [hunk 2]
```diff
@@ -65,6 +65,18 @@ for elty in (Float32, Float64, ComplexF32, ComplexF64, Int)
     @test transpose(W)*v ≈ transpose(F)*v
     @test (W + W) \ v ≈ (2*Matrix(W)) \ v
 
+    # logdet
+    # make sure all matrices are PSD because I don't want complex numbers
+    randpsd(S) = randn(S) |> Q -> Q * Q'
+    W = Woodbury(T, randpsd(size(T)), randpsd(size(T)), randpsd(size(T)))
+    @test logdet(W) ≈ logdet(W.A) + logdet(W.C) - logdet(W.Cp)
+    @test logdet(W) ≈ log(det(W))
```
Comment (timholy) [inline test/woodbury.jl]:
```suggestion
    @test logdet(W) ≈ log(det(W)) ≈ logdet(Array(W))
```

File: test/woodbury.jl [hunk 3]
```diff
@@ -65,6 +65,18 @@ for elty in (Float32, Float64, ComplexF32, ComplexF64, Int)
...
+    lad, s = logabsdet(W)
+    @test lad ≈ logabsdet(W.A)[1] + logabsdet(W.C)[1] - logabsdet(W.Cp)[1]
+    @test s == logabsdet(W.A)[2] * logabsdet(W.C)[2] / logabsdet(W.Cp)[2]
+    @test s == sign(det(W.A)) * sign(det(W.C)) / sign(det(W.Cp))
+    @test lad ≈ log(abs(det(W)))
```
Comment (timholy) [inline test/woodbury.jl]: Maybe this whole block could just be `@test logabsdet(W) ≈ logabsdet(Array(W))`?

**Review by timholy — COMMENTED** [review]
Looks great. In addition to these suggestions, it would be ideal to put corresponding tests into `test/symwoodbury.jl` since the method is defined for `AbstractWoodbury`. Once that's done, I don't see any reason to hold this up any further.

File: test/woodbury.jl (hunk 3 above)
Comment (mileslucas) [reply test/woodbury.jl]: Thats....a good point.

**Comment by mileslucas** [comment]
~ready for re-review~
Any clue why on v1.0 my PSD matrices `randn(n, n) * randn(n, n)'` are getting [negative determinants](https://travis-ci.com/github/mileslucas/WoodburyMatrices.jl/jobs/326201500#L199-L201)?

**Comment by timholy** [comment]
I'd guess it must be from `Cp` which is [computed from your inputs](https://github.com/timholy/WoodburyMatrices.jl/blob/93f7e0778658cef597a1affe06c4650636724be7/src/woodbury.jl#L39). From a quick glance it's not entirely obvious to me that it's PSD.

**Comment by mileslucas** [comment]
My understanding was that I’m just doing the reverse of a cholesky decomposition, which is definitively a PSD matrix. 

The other way I would think of is generate random positive floats as the eigenvalues of the matrix and then creating the matrix. I’ll give that a try, but I’m 
1. Genuinely unsure why my definition is wrong and 
2. Unsure why this only fails on 1.0

Are the operations that define Cp violating the PSDness of my inputs? It doesn’t seem like it since it’s just addition, multiplication, and inversion.

**Comment by timholy** [comment]
Can't look now, have to teach. Maybe a bug on 1.0. If you have a downloaded binary of 1.0 you could replicated the computations for a fixed set of matrices at the REPL.

**Comment by mileslucas** [comment]
I’ll fiddle around a bit, thanks!

**Comment by mileslucas** [comment]
I’m going to close and reopen to trigger the tests again, to see if this is a true bug.

**Comment by mileslucas** [comment]
Okay, unfortunately it appears the previous testing "bug" is just some artifact of the random seeding on julia 1.0 and unlucky floating point precision errors. By adding a few more samples (matrix size 10x10 -> 50x50) this appears to resolve the problem on julia 1.0 without introducing a new one on 1.1-1.4. I played around with the random seed a bit, too, to ensure this could be reproduced with different seeds and all seemed good.

**Comment by timholy** [comment]
Sorry I missed your resubmission here. Thanks for a great contribution!

**Pull request merged**